src/services/sqlObject/core/parameter.ts

```diff
@@ -11,4 +13,65 @@ export class Parameter {
         this.name = name;
         this.type = type;
     }
+
+    /**
+     * 
+     * @param definition of SqlObject.definition
+     * @returns array of parameters
+     */
+    public static async fromDefinition(definition: string) {
+        //get block of header param
+        let definitionHeader = definition.substring(definition.search(/\[[\w]{0,100}\][.]\[[\w]{0,100}\]/), definition.search('AS')).replace(/\[[\w]{0,100}\][.]\[[\w]{0,100}\]/, '').trim();
+        if(definitionHeader[0] == '(') definitionHeader = definitionHeader.substring(1, definitionHeader.length - 1).trim();
+        if(definitionHeader == '') return;
+        definitionHeader += ', @';
+
+        const parameters: Parameter[] = [];
+        let start = -1;
+        //cycle param of header
+        while ((start = definitionHeader.indexOf('@', start + 1)) > -1 && start < definitionHeader.length - 2) {
+            const endLine = definitionHeader.indexOf(', @', start);
+            const paramStr = definitionHeader.substring(start, endLine).trim();
+            const parameter:Parameter = await Parameter?.getParameter(paramStr);
+            parameters.push(parameter);
+        }
+
+        return parameters;
+    }
+
+    private static async getParameter(parameterStr: string) {
+        const parameter: Parameter = {
+            name: await Parameter.getName(parameterStr),
+            type: await Parameter.getType(parameterStr),
+            nullable: parameterStr.indexOf(' NULL') > 0,
+            output: (parameterStr.indexOf(' OUT') > 0 || parameterStr.indexOf(' OUTPUT') > 0),
+            default: await Parameter?.getDefault(parameterStr)
+        };
+        return parameter;
+
+    }
+
+    private static async getName(parameterStr: string) {
+        const start = parameterStr.indexOf('@');// _IdArticolo NVARCHAR(50) = NULL OUT
+        const end = parameterStr.indexOf(' ', start);// @IdArticolo_NVARCHAR(50) = NULL OUT
+        if (start < 0 || start > end) return '';//Something went wrong
+        return parameterStr.substring(start + 1, end);// @|IdArticolo| NVARCHAR(50) = NULL OUT,
+    }
+
+    private static async getType(parameterStr: string) {
+        const start = parameterStr.indexOf(' '); // @IdArticolo_NVARCHAR(50) = NULL OUT
+        const end = parameterStr.indexOf('='); // @IdArticolo NVARCHAR(50, 18)_= NULL OUT
```

Comment:
Parameters don't always end with `=` (see [Stored Procedure Syntax](https://learn.microsoft.com/en-us/sql/t-sql/statements/create-procedure-transact-sql?view=sql-server-ver16#syntax)).
Examples of valid parameters:

- `@IdArticolo NVARCHAR(50) = NULL OUT`
- `@IdArticolo NVARCHAR(50),`
- `@IdArticolo NVARCHAR(50) NULL = 'XXX'`

Every parameter ends with `,` except the last one.

Reply:
I have to check the '=' because i could have this DECIMAL(18, 8) and it wouldn't work, so if there isn't '=' i take all string and remove superfluous things